### ISSUE TITLE
--strict option is ignored by CountInLoopExpression rule

### ISSUE THREAD

**Issue by DannyvdSluijs** [phpmd/phpmd#1020]
- PHPMD version: 2.13.0
- PHP Version: 8.1.19
- Installation type: composer
- Operating System / Distribution & Version: macOS Ventura 13.4.1

### Current Behavior
Using the following code snippet
```php
<?php

declare(strict_types=1);

namespace Test\Bas\TradeWebsite;

class Something
{
    /**
     * @SuppressWarnings(PHPMD.UndefinedVariable)
     * @SuppressWarnings(PHPMD.UnusedLocalVariable)
     * @SuppressWarnings(PHPMD.CountInLoopExpression)
     */
    public function test(): void
    {
        while(count([])) {
            break;
        }

        $y;
    }
}
``` 

when running ` ./vendor/bin/phpmd tests/Something.php ansi phpmd.xml` I get `Found 0 violations and 0 errors in 101ms`
when I add the `--strict` option (`./vendor/bin/phpmd tests/Something.php ansi phpmd.xml --strict`) the output shows:
```text
FILE: ***********/Something.php
------------------------------------------------------------------
 20 | VIOLATION | Avoid unused local variables such as '$y'.
 20 | VIOLATION | Avoid using undefined variables such as '$y' which will lead to PHP notices.
```

### Expected Behavior
I would expect the `CountInLoopExpression` violation to be found as well when running in strict mode.
```text
FILE: ***********/Something.php
------------------------------------------------------------------
 11 | VIOLATION | Avoid using Test\Bas\TradeWebsite\count() function in while loops.
 15 | VIOLATION | Avoid unused local variables such as '$y'.
 15 | VIOLATION | Avoid using undefined variables such as '$y' which will lead to PHP notices.
```

I can see in the XML output the priority of `CountInLoopExpression` is set to 2 and the others are priority 3. This could explain why it doesn't show in strict mode. However as a user of PHMD on the CLI and visitor of the website this behaviour isn't documented. So in my eyes the expected behaviour could also be as is but some clarification in the tool or website is missing. 

### Steps To Reproduce:
Take the above code snippet and run it with `./vendor/bin/phpmd tests/Something.php ansi phpmd.xml --strict`

## Checks before submitting
* [X] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [X] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [X] I have added every step to reproduce the bug.
* [X] If possible I added relevant code examples.
* [X] This issue is about 1 bug and nothing more.
* [X] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

**Comment by tvbeek** [comment]
Thanks for reporting, I have already debugged a part and can confirm that it doesn't do what it need to do. I just need to find out why 😆

**Comment by tvbeek** [comment]
I suspect this impact more rules.

On the AbstractRule class we have:
```php
    protected function applyOnClassMethods(AbstractTypeNode $node)
    {
        foreach ($node->getMethods() as $method) {
            if ($method->hasSuppressWarningsAnnotationFor($this)) {
                continue;
            }

            $this->apply($method);
        }
    }
```

That is used in the apply method from the CountInLoopExpression. And added for #677 

I'm not yet sure why this rule has another method to add the violations. But it is a next step in fixing this issue 😃
